blog/2023-07-13-reproducibility-designer.md

```diff
@@ -0,0 +1,41 @@
+---
+author: Antonin Delpeuch
+title: "Designer seeked for OpenRefine's reproducibility improvements"
+---
+
+OpenRefine is hiring a designer to help shape our improvements to OpenRefine's operation history and test those changes with user panels.
+<!--truncate-->
+
+[OpenRefine](https://openrefine.org/) is a powerful tool to clean messy data, popular in a diverse range of communities. It has been serving the needs of journalists, librarians, Wikipedians, scientists for more than ten years and is taught in many curricula and workshops around the world. OpenRefine is a fiscally sponsored project of [Code for Science & Society Inc](https://codeforscience.org/), a 501(c)(3) charitable organization in the USA.
+
+We are currently working on improving the reproducibility of OpenRefine workflows as part of [a project funded by the Essential Open Source Software for Science programme](https://chanzuckerberg.com/eoss/proposals/improving-openrefines-reproducibility/).
+This consists in improving the user experience around OpenRefine's operation history, which tracks all transformations applied on the original data. This work aims to address a range of long standing issues, such as:
+- limitations of the Undo/Redo functionality, such as the lack of ability to undo an operation in the middle of the history list ([#183](https://github.com/OpenRefine/OpenRefine/issues/183)) or unintentional loss of work after using the undo feature and
+  applying a new operation ([#369](https://github.com/OpenRefine/OpenRefine/issues/369), [#3184](https://github.com/OpenRefine/OpenRefine/issues/3184))
+- the difficulty in reusing and sharing OpenRefine workflows, due to the lack of error handling when applying series of operations and the difficulty to adapt a workflow to a new project (for instance, with different column names)
+- the lack of support for running OpenRefine workflows as part of larger pipelines, without using the web UI interactively. This covers for instance the lack of import metadata in the JSON export of the history
+  ([#460](https://github.com/OpenRefine/OpenRefine/issues/460)).
+
+See the corresponding [GitHub project](https://github.com/orgs/OpenRefine/projects/6) for more issues in the scope of this project.
+
+We are looking for a part-time designer who will
+* design changes to the user interface to support those improvements;
+* validate those choices by conducting user testing sessions;
+* help prioritize follow-up improvements based on user feedback.
+
+Requirements: 
+* Experience with designing and testing software
+* Ability to work in a fully remote environment
+* Good communication skills and fluency in English
+
+Nice to have:
+* Familiarity with OpenRefine
```

Comment:
I'm struggling to see how this can be optional for a successful candidate. They'd at least need strong familiarity with one or more of other data wrangling software, workflow languages, or related topics.

Reply:
Yes, I also see it as a rather important skill, but my intention with classifying it as an optional skill is to broaden a bit the pool of potential applicants. I thought it's worth being open minded about the skills that someone would be bringing to the table: I can imagine someone with sufficient work experience and a solid methodology can be a good match for this role even if they are not familiar yet with the tool itself, as they'd be able to rely on the rest of the team for subtle product decisions but be more of a driving force around the overall design and testing process.

But overall I don't have strong feelings about how this should be worded / classified.

Reply:
I kind of agree that familiarity with OpenRefine or some other data wrangling software should be required and not just a nice to have except the role makes room for a duration of onboarding because from experience, OpenRefine is not really the easiest of tools for a designer to get familiar with in a short time frame and especially on their own and this project seldom feels like a beginner's forte.

Reply:
I've moved this point to the requirements section, making it a bit lighter by accepting familiarity with other data cleaning tools as well.

Reply:
Let's be honest - how many people in the existing OR community are designers? If this is a requirement, you will get very few candidates and those would be probably developers with interest in design, rather than designers per se. But maybe few candidates is OK, if you have limited time and scope for recruiting.

Reply:
> I've moved this point to the requirements section, making it a bit lighter by accepting familiarity with other data cleaning tools as well.

Thanks. I'm happy with broadening further this to include other relevant experience such as workflow managers (Galaxy, Nextflow, Airflow, CWL) in addition to data wrangling tools.